internal/ports/convert.go

```diff
@@ -84,45 +71,83 @@ func (c *Client) Convert() *cobra.Command {
+				_, _, err := map[bool]func(string) (*metal.Port, *http.Response, error){
 					true:  convToL2,
 					false: convToL3,
 				}[layer2](portID)
 				if err != nil {
 					return fmt.Errorf("failed to change port network mode: %w", err)
 				}
 			}
-			listOpts := c.Servicer.ListOptions(nil, nil)
 
-			getOpts := &packngo.GetOptions{Includes: listOpts.Includes, Excludes: listOpts.Excludes}
-			port, _, err := c.PortService.Get(portID, getOpts)
+			port, _, err := c.PortService.FindPortById(context.Background(), portID).
+				Include(c.Servicer.Includes(nil)).
+				Execute()
 			if err != nil {
 				return fmt.Errorf("Could not get Port: %w", err)
 			}
 
 			data := make([][]string, 1)
 
-			data[0] = []string{port.ID, port.Name, port.Type, port.NetworkType, port.Data.MAC, strconv.FormatBool(port.Data.Bonded)}
+			data[0] = []string{port.GetId(), port.GetName(), string(port.GetType()), string(port.GetNetworkType()), port.Data.GetMac(), strconv.FormatBool(port.Data.GetBonded())}
 			header := []string{"ID", "Name", "Type", "Network Type", "MAC", "Bonded"}
 
 			return c.Out.Output(port, header, &data)
 		},
 	}
 
-	retrievePortCmd.Flags().StringVarP(&portID, "port-id", "i", "", "The UUID of a port.")
-	retrievePortCmd.Flags().BoolVarP(&bonded, "bonded", "b", false, "Convert to layer-2 bonded.")
-	retrievePortCmd.Flags().BoolVarP(&bulk, "bulk", "", false, "Affect both ports in a bond.")
-	retrievePortCmd.Flags().BoolVarP(&layer2, "layer2", "2", false, "Convert to layer-2 unbonded.")
-	retrievePortCmd.Flags().BoolVarP(&force, "force", "f", false, "Force conversion to layer-2 bonded.")
-	retrievePortCmd.Flags().BoolVarP(&ipv4, "public-ipv4", "4", false, "Convert to layer-2 bonded with public IPv4.")
-	retrievePortCmd.Flags().BoolVarP(&ipv6, "public-ipv6", "6", false, "Convert to layer-2 bonded with public IPv6.")
+	convertPortCmd.Flags().StringVarP(&portID, "port-id", "i", "", "The UUID of a port.")
+	convertPortCmd.Flags().BoolVarP(&bonded, "bonded", "b", false, "Convert to layer-2 bonded.")
+	convertPortCmd.Flags().BoolVarP(&bulk, "bulk", "", false, "Affect both ports in a bond.")
+	convertPortCmd.Flags().BoolVarP(&layer2, "layer2", "2", false, "Convert to layer-2 unbonded.")
+	convertPortCmd.Flags().BoolVarP(&force, "force", "f", false, "Force conversion to layer-2 bonded.")
+	convertPortCmd.Flags().BoolVarP(&ipv4, "public-ipv4", "4", false, "Convert to layer-2 bonded with public IPv4.")
+	convertPortCmd.Flags().BoolVarP(&ipv6, "public-ipv6", "6", false, "Convert to layer-2 bonded with public IPv6.")
+
+	return convertPortCmd
+}
+
+func portBondingHandler(cmd *cobra.Command, c *Client, portId string) error {
+	if cmd.Flag("bonded").Changed {
+		_, _, err := c.PortService.BondPort(context.Background(), portId).
+			Include(c.Servicer.Includes(nil)).
+			Execute()
+		return err
+	}
 
-	return retrievePortCmd
+	_, _, err := c.PortService.DisbondPort(context.Background(), portId).
```

Comment:
The err returned is:
```
Error: failed to change port bonding: 422 Unprocessable Entity
```

The more actionable error message is not being surfaced:
```
{"errors":["Bond operations on port bond0 are not supported while the Hybrid bonded mode is enabled"]}
```

We may have been inadvertantly sliding backwards on error reporting throughout the metal-go conversion.
We should be returning something more like what was returned with packngo:
https://github.com/packethost/packngo/blob/master/packngo.go#L78-L81

(cc @ctreatma)

Reply:
Yes, this is a very valid point. Even while testing I have received errors such as `422 Unprocessable Entity` really couldnt understand what exactly is missing. Had to make some trial and errors